### PR TITLE
updated avatar component name, added ability to change username (only works server side for now), updated signin page, updated Footer

### PR DESCRIPTION
**Pull request by rsandb**
A bunch of these commits were already added so I'm not sure why they're showing up here. The only one that I'm requesting is "[de86903](https://github.com/LAION-AI/Open-Assistant/pull/91/commits/de869033bf9edf3c051d01c264637b3d286a65ae)"

### CODE REVIEW

File: website/src/pages/account/index.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import Head from "next/head";
+import Link from "next/link";
+import Router from "next/router";
+import React, { useEffect, useState } from "react";
+import { getSession, useSession } from "next-auth/react";
+import { Button, Input, InputGroup, Stack } from "@chakra-ui/react";
+import { Footer } from "../../components/Footer";
```
Comment (fozziethebeat) [inline website/src/pages/account/index.tsx:7]: Can you use absolute imports for these? (`src/components/Footer`)

File: website/src/pages/account/edit.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import Head from "next/head";
+import Link from "next/link";
+import Router from "next/router";
+import React, { useState } from "react";
+import { getSession, useSession } from "next-auth/react";
+import { PrismaClient, User } from "@prisma/client";
+import { Button, Input, InputGroup, Stack } from "@chakra-ui/react";
+import { Footer } from "../../components/Footer";
```
Comment (fozziethebeat) [inline website/src/pages/account/edit.tsx:8]: Can you use absolute imports here?

File: website/src/pages/account/edit.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import Head from "next/head";
+import Link from "next/link";
+import Router from "next/router";
+import React, { useState } from "react";
+import { getSession, useSession } from "next-auth/react";
+import { PrismaClient, User } from "@prisma/client";
+import { Button, Input, InputGroup, Stack } from "@chakra-ui/react";
```
Comment (fozziethebeat) [inline website/src/pages/account/edit.tsx:7]: orderwise I like to treat `@` imports as first, can you put this first?

File: website/src/pages/account/edit.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import styles from "../styles/Home.module.css";
+
+export default function Account() {
+  const { data: session } = useSession();
+  const [username, setUsername] = useState("");
+  const updateUser = async (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    try {
+      const body = { username };
+      await fetch("../api/username", {
```
Comment (fozziethebeat) [inline website/src/pages/account/edit.tsx:20]: Just to ask, why is the url `../api/username`? Shouldn't `/api/username` work?

**Review by fozziethebeat — CHANGES_REQUESTED** [review]

**Comment by fozziethebeat** [comment]
Will approve after I see the conflicts are merged in and the small style changes are made.

File: website/src/pages/account/edit.tsx (hunk 4 above)
Comment (rsandb) [reply website/src/pages/account/edit.tsx]: Yeah it does, didn't catch that thanks!

File: website/src/middleware.ts [hunk 5]
```diff
@@ -4,5 +4,5 @@ export { default } from "next-auth/middleware";
  * Guards all pages under `/grading` and redirects them to the sign in page.
  */
 export const config = {
-  matcher: ["/create/:path*", "/evaluate/:path*"],
+  matcher: ["/create/:path*", "/evaluate/:path*", "/grading/:path*", "/account/:path*", "/summarize/:path*"],
```
Comment (AbdBarho) [inline website/src/middleware.ts:7]: /grading
and
/summarize
are deprecated, you can remove them.

File: website/src/pages/404.tsx [hunk 6]
```diff
@@ -5,39 +5,15 @@ import Head from "next/head";
...
       <Head>
-        <title>Open Assistant</title>
-        <meta
-          name="description"
-          content="Conversational AI for everyone. An open source project to create a chat enabled GPT LLM run by LAION and contributors around the world."
-        />
+        <title>404 - Open Assistant</title>
+        <meta name="404" content="Sorry, this page doesn't exist." />
       </Head>
       <Header />
```
Comment (AbdBarho) [inline website/src/pages/404.tsx:14]: could you please also remove the header and footer here?

File: website/src/pages/_app.tsx [hunk 7]
```diff
@@ -40,7 +40,7 @@ function MyApp({ Component, pageProps: { session, ...pageProps } }: AppPropsWith
 
   return (
     <ChakraProvider theme={theme}>
-      <SessionProvider session={session}>{page}</SessionProvider>
+      <SessionProvider session={pageProps.session}>{page}</SessionProvider>
```
Comment (AbdBarho) [inline website/src/pages/_app.tsx:43]: why `pageProps.session`? is `session` not enough?

File: website/src/pages/account/edit.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+import { getSession, useSession } from "next-auth/react";
+import { PrismaClient, User } from "@prisma/client";
+import { Button, Input, InputGroup, Stack } from "@chakra-ui/react";
+import Head from "next/head";
+import Link from "next/link";
+import Router from "next/router";
+import { Footer } from "src/components/Footer";
+import { Header } from "src/components/Header";
```
Comment (AbdBarho) [inline website/src/pages/account/edit.tsx:9]: could you remove unused imports?

File: website/src/pages/account/index.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+import { Header } from "../../components/Header";
```
Comment (AbdBarho) [inline website/src/pages/account/index.tsx:8]: could you remove unused imports?

File: website/src/pages/api/username.tsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+// Required fields in body: title
+// Optional fields in body: content
+export default async function handle(req, res) {
+  const { username } = req.body;
+  const { email } = req.body;
+
+  const session = await getSession({ req });
+  const result = await prisma.user.update({
+    where: {
+      email: session?.user?.email,
```
Comment (AbdBarho) [inline website/src/pages/api/username.tsx:15]: is it ok if any of `session?.user?.email` is undefined?

If not, remove all question marks.

**Review by AbdBarho — COMMENTED** [review]
Thanks!

**Review by AbdBarho — APPROVED** [review]
Thanks!

LGTM

We will have to wait for @fozziethebeat to approve.

**Review by k-nearest-neighbor — APPROVED** [review]
Looks great!

**Review by fozziethebeat — APPROVED** [review]

**Pull request merged**